### DISCUSSION TITLE
Replace If activity with a custom one

### DISCUSSION THREAD

**Discussion by davidediluccio97** [elsa-workflows/elsa-core#4083] [Q&A] [answered]
I'm using Elsa 2 and I want to replace the **If activity** with a custom one that I've made,  but I'm not able to remove the original If activity. I've tried to use the `RemoveActivity` method of `ElsaOptionsBuilder` but it doesn't work.

Can anyone help me?

Thanks

**Comment by sfmskywalker** [comment, marked as answer]
There's currently no elegant way to control exactly what activities to add, without adding them yourself manually, but maybe that's OK for your scenario.

If so, then you can call `NoCoreActivities()` on the Elsa options builder.
When you do, you'll need to add the activities you **do* want, yourself.

For Elsa 3, there will be better support to "opt out" of certain activities.

**Reply by davidediluccio97** [reply]
Thank you for the answer @sfmskywalker! I'll use the `NoCoreActivities()` method.
